dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py

```diff
@@ -996,3 +996,63 @@ def test_diff(self):
             str(exc.exception),
             "Unsupported operand type(s) for diff: 'TestColumnWProps' and" " 'str'",
         )
+
+        # PSI same distribution test
+        other1, other2 = TestColumnWProps(), TestColumnWProps()
+        other1.match_count = 55
+        other1._stored_histogram = {
+            "total_loss": 0,
+            "current_loss": 0,
+            "suggested_bin_count": 10,
+            "histogram": {
+                "bin_counts": np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+                "bin_edges": np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
+            },
+        }
+
+        other2.match_count = 550
+        other2._stored_histogram = {
+            "total_loss": 0,
+            "current_loss": 0,
+            "suggested_bin_count": 10,
+            "histogram": {
+                "bin_counts": np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]) * 10,
+                "bin_edges": np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
+            },
+        }
+
+        expected_psi_value = 0
+        psi_value = other1._calculate_psi(
+            self_match_count=other1.match_count,
+            self_histogram=other1._stored_histogram["histogram"],
+            other_match_count=other2.match_count,
+            other_histogram=other2._stored_histogram["histogram"],
+        )
+        self.assertEquals(expected_psi_value, psi_value)
+
+        # PSI min_min_edge == max_max_edge
+        other1, other2 = TestColumnWProps(), TestColumnWProps()
+        other1.match_count = 10
+        other1._stored_histogram = {
+            "total_loss": 0,
+            "current_loss": 0,
+            "suggested_bin_count": 10,
+            "histogram": {"bin_counts": np.array([10]), "bin_edges": np.array([1, 1])},
+        }
+
+        other2.match_count = 20
+        other2._stored_histogram = {
+            "total_loss": 0,
+            "current_loss": 0,
+            "suggested_bin_count": 10,
+            "histogram": {"bin_counts": np.array([20]), "bin_edges": np.array([1, 1])},
+        }
+
+        expected_psi_value = 0
+        psi_value = other1._calculate_psi(
+            self_match_count=other1.match_count,
+            self_histogram=other1._stored_histogram["histogram"],
+            other_match_count=other2.match_count,
+            other_histogram=other2._stored_histogram["histogram"],
+        )
+        self.assertEquals(expected_psi_value, psi_value)
```

Comment:
There might be 1 more edge case.

when one requires regen and the other doesn't and the transpose. Can use the same histograms, just flipped positions.

Thoughts?

Reply:
pushing

Reply:
@taylorfturner is adding.